src/components/TagButton.tsx

```diff
@@ -64,23 +62,24 @@ export const TagButton: (props: TagButtonProps) => JSX.Element = ({
     e: React.KeyboardEvent<HTMLDivElement>,
     item: ListItem
   ) => {
-    const targetTag = e.target as HTMLTextAreaElement;
-    const targetTagTitle = targetTag.value;
-    if (e.key === "Enter" && targetTagTitle) {
-      const tagAlreadyAssigned = isTagNew(targetTagTitle, item.tags);
-      const tagAlreadyKnown = isTagNew(targetTagTitle, knownTags);
+    const tag = e.target as HTMLTextAreaElement;
+    const tagTitle = tag.value;
+    if (e.key === "Enter" && tagTitle) {
+      const tagAlreadyAssigned = isTagNew(tagTitle, item.tags);
+      const tagAlreadyKnown = isTagNew(tagTitle, knownTags);
       if (!tagAlreadyAssigned) {
         if (tagAlreadyKnown) {
-          tagAdded(item, tagAlreadyKnown);
+          dispatch(assignTagToItem(tagAlreadyKnown, item));
           setErrorTag(errorTags.empty);
         } else {
-          const tagIsNew = updateKnownTags(targetTagTitle);
-          if (tagIsNew) {
-            tagAdded(item, tagIsNew);
-            setErrorTag(errorTags.empty);
-          } else {
-            setErrorTag(errorTags.wrong);
-          }
+          dispatch(updateKnownTags(tagTitle));
```

Comment:
Right after this action, another action should be dispatched:
`dispatch(assignTagToItem(X, item))`
... where X stands for previously created tag, which I can't seem to get neither from filtering it from the knownTags, nor reusing it from the return of the previous action.

Would you @erikvanek help a no-bro out?

Reply:
how about doing something like `const newTag = updateKnownTags(tagTitle)` and then `dispatch(newTag); dispatch(assignTagToItem(newTag.payload.tag, item))`? Not sure if it works though.